Getting compile error when I use silius as library

```
[dependencies]
silius-bundler = { git = "https://github.com/zsluedem/silius.git" }
silius-contracts = { git = "https://github.com/zsluedem/silius.git" }
silius-grpc = { git = "https://github.com/zsluedem/silius.git" }
silius-mempool = { git = "https://github.com/zsluedem/silius.git" }
silius-p2p = { git = "https://github.com/zsluedem/silius.git" }
silius-primitives = { git = "https://github.com/zsluedem/silius.git" }
silius-rpc = { git = "https://github.com/zsluedem/silius.git" }
```

Got error
```
error[E0432]: unresolved import `revm_primitives::B160`
   --> /home/will/.cargo/git/checkouts/reth-36d3ea1d1152b20c/aa6f2cb/crates/rlp/src/decode.rs:255:58
    |
255 |     use revm_primitives::{ruint::aliases::U128 as RU128, B160, B256, U256 as RU256};
    |                                                          ^^^^ no `B160` in the root

error[E0432]: unresolved import `revm_primitives::B160`
   --> /home/will/.cargo/git/checkouts/reth-36d3ea1d1152b20c/aa6f2cb/crates/rlp/src/encode.rs:216:58
    |
216 |     use revm_primitives::{ruint::aliases::U128 as RU128, B160, B256, U256 as RU256};
    |                                                          ^^^^ no `B160` in the root

error[E0432]: unresolved import `revm_primitives::B160`
 --> /home/will/.cargo/git/checkouts/reth-36d3ea1d1152b20c/aa6f2cb/crates/storage/codecs/src/lib.rs:3:23
  |
3 | use revm_primitives::{B160 as H160, B256 as H256, U256};
  |                       ^^^^^^^^^^^^ no `B160` in the root

For more information about this error, try `rustc --explain E0432`.
error: could not compile `reth-codecs` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
error: the `Self` constructor can only be used with tuple or unit structs
   --> /home/will/.cargo/git/checkouts/reth-36d3ea1d1152b20c/aa6f2cb/crates/rlp/src/decode.rs:261:48
    |
261 |                     Decodable::decode(buf).map(Self)
    |                                                ^^^^
...
267 |     fixed_hash_impl!(B160);
    |     ---------------------- in this macro invocation
    |
    = note: this error originates in the macro `fixed_hash_impl` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `reth-rlp` (lib) due to 3 previous errors

```

--- a/bin/silius/Cargo.toml
+++ b/bin/silius/Cargo.toml
@@ -16,7 +16,7 @@ homepage = "https://github.com/silius-rs/silius/tree/main/bin/silius"
 silius-bundler = { workspace = true }
 silius-contracts = { workspace = true }
 silius-grpc = { workspace = true }
-silius-mempool = { workspace = true }
+silius-mempool = { workspace = true, features = ["mdbx"] }
 silius-p2p = { workspace = true }
 silius-primitives = { workspace = true }
 silius-rpc = { workspace = true }

--- a/crates/mempool/Cargo.toml
+++ b/crates/mempool/Cargo.toml
@@ -21,8 +21,8 @@ alloy-chains = { workspace = true }
 ethers = { workspace = true }
 
 # reth
-reth-db = { git = "https://github.com/paradigmxyz/reth.git", rev = "aa6f2cb0610fb4fa0926b42cfed7f8ff51e0db8a" }
-reth-libmdbx = { git = "https://github.com/paradigmxyz/reth.git", rev = "aa6f2cb0610fb4fa0926b42cfed7f8ff51e0db8a" }
+reth-db = { git = "https://github.com/paradigmxyz/reth.git", rev = "aa6f2cb0610fb4fa0926b42cfed7f8ff51e0db8a", optional = true }
+reth-libmdbx = { git = "https://github.com/paradigmxyz/reth.git", rev = "aa6f2cb0610fb4fa0926b42cfed7f8ff51e0db8a", optional = true }
 
 # async
 async-trait = { workspace = true }
@@ -50,3 +50,6 @@ silius-primitives = { workspace = true, features = ["test-utils"] }
 
 # misc
 tempdir = "0.3.7"
+
+[features]
+mdbx = ["dep:reth-db", "dep:reth-libmdbx"]

--- a/crates/mempool/src/lib.rs
+++ b/crates/mempool/src/lib.rs
@@ -2,6 +2,7 @@
 #![allow(dead_code)]
 
 mod builder;
+#[cfg(feature = "mdbx")]
 mod database;
 mod memory;
 mod mempool;
@@ -11,6 +12,7 @@ mod utils;
 pub mod validate;
 
 pub use builder::UoPoolBuilder;
+#[cfg(feature = "mdbx")]
 pub use database::{
     init_env,
     tables::{

--- a/crates/mempool/src/mempool.rs
+++ b/crates/mempool/src/mempool.rs
@@ -1,3 +1,4 @@
+#[cfg(feature = "mdbx")]
 use crate::DBError;
 use ethers::{
     abi::AbiEncode,
@@ -18,21 +19,25 @@ pub fn mempool_id(ep: &Address, chain_id: &U256) -> MempoolId {
 
 #[derive(Debug)]
 pub enum MempoolError {
+    #[cfg(feature = "mdbx")]
     DBError(DBError),
 }
 
+#[cfg(feature = "mdbx")]
 impl From<DBError> for MempoolError {
     fn from(e: DBError) -> Self {
         Self::DBError(e)
     }
 }
 
+#[cfg(feature = "mdbx")]
 impl From<reth_db::Error> for MempoolError {
     fn from(e: reth_db::Error) -> Self {
         Self::DBError(e.into())
     }
 }
 
+#[cfg(feature = "mdbx")]
 impl From<MempoolError> for DBError {
     fn from(e: MempoolError) -> Self {
         match e {

--- a/crates/mempool/src/reputation.rs
+++ b/crates/mempool/src/reputation.rs
@@ -1,4 +1,6 @@
-use crate::{mempool::ClearOp, DBError};
+use crate::mempool::ClearOp;
+#[cfg(feature = "mdbx")]
+use crate::DBError;
 use ethers::types::{Address, Bytes, U256};
 use parking_lot::RwLock;
 use silius_primitives::{
@@ -9,6 +11,7 @@ use std::{fmt::Debug, ops::Deref, sync::Arc};
 
 #[derive(Debug)]
 pub enum ReputationOpError {
+    #[cfg(feature = "mdbx")]
     DBError(DBError),
     ReputationError(ReputationError),
 }
@@ -19,12 +22,14 @@ impl From<ReputationError> for ReputationOpError {
     }
 }
 
+#[cfg(feature = "mdbx")]
 impl From<DBError> for ReputationOpError {
     fn from(value: DBError) -> Self {
         Self::DBError(value)
     }
 }
 
+#[cfg(feature = "mdbx")]
 impl From<reth_db::Error> for ReputationOpError {
     fn from(value: reth_db::Error) -> Self {
         Self::DBError(DBError::DBInternalError(value))

--- a/examples/storage/Cargo.toml
+++ b/examples/storage/Cargo.toml
@@ -14,7 +14,7 @@ homepage = "https://github.com/silius-rs/silius/tree/main/examples/storage"
 [dependencies]
 # silius dependencies
 silius-contracts = { workspace = true }  # replace with git url: git = "https://github.com/silius-rs/silius.git"
-silius-mempool = { workspace = true }    # replace with git url: git = "https://github.com/silius-rs/silius.git"
+silius-mempool = { workspace = true, features = ["mdbx"] }    # replace with git url: git = "https://github.com/silius-rs/silius.git"
 silius-primitives = { workspace = true } # replace with git url: git = "https://github.com/silius-rs/silius.git"
 
 # eth

